### PR TITLE
fix(database-conf): detail BDM grants oracle

### PR DESCRIPTION
**Pull request by marielle-spiteri**
* add clear explanation on grants for the BDM DB

### CODE REVIEW

File: modules/ROOT/pages/database-configuration.adoc [hunk 1]
```diff
@@ -61,9 +61,11 @@ Download the driver compatible with Oracle 12.2.0.1: https://www.oracle.com/data
 
 The first step in configuring Bonita with your RDBMS is to create the new databases (i.e. schemas): one for engine data, and another one, optionally, if you use business data.
 
-To do so, you need a RDBMS user account that has sufficient privileges (i.e. privileges to create new schema).
+WARNING: If you use different users for the engine database and for the business data database, then you must *grant the following privileges to both users*.
```
Comment (NathalieC) [inline modules/ROOT/pages/database-configuration.adoc]: This warning should come below the first sentence  that mentions the users.

File: modules/ROOT/pages/database-configuration.adoc [hunk 2]
```diff
@@ -61,9 +61,11 @@ Download the driver compatible with Oracle 12.2.0.1: https://www.oracle.com/data
 
 The first step in configuring Bonita with your RDBMS is to create the new databases (i.e. schemas): one for engine data, and another one, optionally, if you use business data.
 
-To do so, you need a RDBMS user account that has sufficient privileges (i.e. privileges to create new schema).
+WARNING: If you use different users for the engine database and for the business data database, then you must *grant the following privileges to both users*. 
 
-Also, note that the owner of the new schemas must own the following privileges:
+To create each new database, you need a RDBMS user account that has sufficient privileges (i.e. privileges to create new schema).
```
Comment (NathalieC) [inline modules/ROOT/pages/database-configuration.adoc]:
```suggestion
To create each new database, you need a RDBMS user account that has sufficient privileges (i.e. privileges to create new schemas).
```

File: modules/ROOT/pages/database-configuration.adoc (hunk 1 above)
Comment (NathalieC) [inline modules/ROOT/pages/database-configuration.adoc]:
```suggestion
WARNING: If you use different RDBMS users for the Bonita Engine database and for the business data database, then you must *grant the following privileges to both users*. 
```

Comment (NathalieC) [inline modules/ROOT/pages/database-configuration.adoc]: This sentence talks about "the following privileges". 
Row 68 also talks about "the following privileges".
But this one is about the RDBMS users, and the other one is about a database owner. 
- Are those users the same? If so, I'd be in favor of always referring to them with the same descriptor.
- Can we talk only once about "the following privileges" (the one time that is closer to the description of the privileges)?

File: modules/ROOT/pages/database-configuration.adoc [hunk 3]
```diff
@@ -187,7 +189,12 @@ SQL query >
 @/u01/app/oracle/product/12.2.0/dbhome_1/rdbms/admin/xaview.sql
 ----
 +
-The following queries must be done for each new user: i.e. one user = one database schema.
+The following queries must be done for each new user: i.e. one user = one database schema.  
++
+WARNING: The queries must be done for *both* the bonita engine *and* the business data database users. +
+For example, if the users for the Bonita BPM and BDM databases are `bonita` and `business_data` respectively (the actual values correspond to the ones set in the properties `db.user` and the `bdm.db.user` defined in the `<TOMCAT_HOME>/setup/database.properties`), you need to run the queries for both users.
```
Comment (NathalieC) [inline modules/ROOT/pages/database-configuration.adoc]:
```suggestion
For example, if the users for the Bonita Engine and business data databases are `bonita` and `business_data` respectively (the actual values correspond to the ones set in the properties `db.user` and the `bdm.db.user` defined in the `<TOMCAT_HOME>/setup/database.properties`), you need to run the queries for both users. 
```

File: modules/ROOT/pages/database-configuration.adoc [hunk 4]
```diff
@@ -187,7 +189,12 @@ SQL query >
 @/u01/app/oracle/product/12.2.0/dbhome_1/rdbms/admin/xaview.sql
 ----
 +
-The following queries must be done for each new user: i.e. one user = one database schema.
+The following queries must be done for each new user: i.e. one user = one database schema.  
++
+WARNING: The queries must be done for *both* the bonita engine *and* the business data database users. +
+For example, if the users for the Bonita BPM and BDM databases are `bonita` and `business_data` respectively (the actual values correspond to the ones set in the properties `db.user` and the `bdm.db.user` defined in the `<TOMCAT_HOME>/setup/database.properties`), you need to run the queries for both users. 
++
+*For the bonita engine database user (e.g. _bonita_):*
```
Comment (NathalieC) [inline modules/ROOT/pages/database-configuration.adoc]:
```suggestion
*For the Bonita Engine database user (e.g. _bonita_):*
```

File: modules/ROOT/pages/database-configuration.adoc [hunk 5]
```diff
@@ -187,7 +189,12 @@ SQL query >
 @/u01/app/oracle/product/12.2.0/dbhome_1/rdbms/admin/xaview.sql
 ----
 +
-The following queries must be done for each new user: i.e. one user = one database schema.
+The following queries must be done for each new user: i.e. one user = one database schema.  
++
+WARNING: The queries must be done for *both* the bonita engine *and* the business data database users. +
```
Comment (NathalieC) [inline modules/ROOT/pages/database-configuration.adoc]:
```suggestion
WARNING: The queries must be done for *both* the Bonita Engine *and* the business data database users. +
```

File: modules/ROOT/pages/database-configuration.adoc [hunk 6]
```diff
@@ -72,7 +74,7 @@ Also, note that the owner of the new schemas must own the following privileges:
 [NOTE]
 ====
 
-CREATE TABLE and CREATE INDEX privileges are not required after first start in normal use. +
+CREATE TABLE and CREATE INDEX privileges are not required for the *engine database* user after first start in normal use (however, they will still be required by the *business data database* user even beyond the first deployment of a BDM). +
```
Comment (NathalieC) [inline modules/ROOT/pages/database-configuration.adoc]: We cannot say that the owners of both databases must have the same privileges and then say that some privileges are not required for one of them.
For more clarity, I would suggest to first describe the privileges of the Bonita Engine database owner, and then the privileges of the business data database.

File: modules/ROOT/pages/database-configuration.adoc [hunk 7]
```diff
@@ -61,9 +61,11 @@ Download the driver compatible with Oracle 12.2.0.1: https://www.oracle.com/data
 
 The first step in configuring Bonita with your RDBMS is to create the new databases (i.e. schemas): one for engine data, and another one, optionally, if you use business data.
 
-To do so, you need a RDBMS user account that has sufficient privileges (i.e. privileges to create new schema).
+WARNING: If you use different users for the engine database and for the business data database, then you must *grant the following privileges to both users*. 
 
-Also, note that the owner of the new schemas must own the following privileges:
+To create each new database, you need a RDBMS user account that has sufficient privileges (i.e. privileges to create new schema).
+
+Next, the owner of each new schema (i.e. engine data *and* business data) must own the following privileges:
```
Comment (NathalieC) [inline modules/ROOT/pages/database-configuration.adoc]:
```suggestion
Next, the owner of each new schema (i.e. Bonita Engine data *and* business data) must own the following privileges:
```

File: modules/ROOT/pages/database-configuration.adoc (hunk 6 above)
Comment (NathalieC) [inline modules/ROOT/pages/database-configuration.adoc]:
```suggestion
CREATE TABLE and CREATE INDEX privileges are not required for the *Bonita Engine database* user after first start in normal use (however, they will still be required by the *business data database* user even beyond the first deployment of a BDM). +
```

**Review by NathalieC — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/database-configuration.adoc (hunk 6 above)
Comment (educhastenier) [reply modules/ROOT/pages/database-configuration.adoc]: And instead of **Engine**, I would say **Runtime**

**Review by educhastenier — APPROVED** [review]
Just one global remark: in this page, the naming of the "Bonita" / "Engine" database is not always named the same way.
Please homogeneize it (have a look at its name on other pages as well)

File: modules/ROOT/pages/database-configuration.adoc (hunk 1 above)
Comment (marielle-spiteri) [reply modules/ROOT/pages/database-configuration.adoc]: Change done

Comment (marielle-spiteri) [reply modules/ROOT/pages/database-configuration.adoc]: * I tried to hamronize the user descriptor throughout the page where it made sense. 
* removed following from the first sentence.

File: modules/ROOT/pages/database-configuration.adoc (hunk 6 above)
Comment (marielle-spiteri) [reply modules/ROOT/pages/database-configuration.adoc]: * I agree with @NathalieC , this note is not useful and there are too many conditions for these priviledges aren't required. But 90% of the time, they are required (except when "first start in normal use" which is not clear to me, please define "first start in normal use"...). Let's say they are always required so we cover all the use cases. And we can remove that note. 
* Regarding the Engine vs Runtime, is that the new word to be used in the documentation? If that is the case, then we need to review all the pages that talk about the Engine. I say Engine is fine for me.

Comment (educhastenier) [reply modules/ROOT/pages/database-configuration.adoc]: @mariellebonita Fine to me.
I let you do this last removal and merge the PR like that (the rest is fine to me).

Comment (educhastenier) [reply modules/ROOT/pages/database-configuration.adoc]: @mariellebonita I removed the note

**Pull request merged**